kratos/linear_solvers/skyline_lu_factorization_solver.h

```diff
@@ -573,211 +503,122 @@ class SkylineLUFactorizationSolver
+
+        KRATOS_TRY
+        mFactorization.clear();
+        mFactorization.copyFromCSRMatrix(rLhs);
+        mFactorization.factorize();
+        KRATOS_CATCH("")
+    }
 
+    /// @copydoc BaseType::PerformSolutionStep
+    bool PerformSolutionStep(SparseMatrixType& rLhs,
+                             VectorType& rSolution,
+                             VectorType& rRhs) override
+    {
+        KRATOS_TRY
+        const auto system_size = TSparseSpaceType::Size(rSolution);
+        mFactorization.backForwardSolve(system_size, rRhs, rSolution);
         return true;
+        KRATOS_CATCH("")
     }
 
-    /** Multi solve method for solving a set of linear systems with same coefficient matrix.
-    Solves the linear system Ax=b and puts the result on SystemVector& rX.
-    rX is also th initial guess for iterative methods.
-    @param rA. System matrix
-    @param rX. Solution vector.
-    @param rB. Right hand side vector.
-    */
+    /// @copydoc BaseType::Solve(SparseMatrixType6,VectorType&,VectorType&)
+    bool Solve(SparseMatrixType& rA, VectorType& rX, VectorType& rB) override
+    {
+        KRATOS_TRY
+        this->InitializeSolutionStep(rA, rX, rB);
+        const auto status = this->PerformSolutionStep(rA, rX, rB);
+        this->FinalizeSolutionStep(rA, rX, rB);
+        return status;
+        KRATOS_CATCH("")
+    }
+
+    /// @copydoc BaseType::Solve(SparseMatrixType6,DenseMatrixType&,DenseMatrixType&)
     bool Solve(SparseMatrixType& rA, DenseMatrixType& rX, DenseMatrixType& rB) override
     {
+        KRATOS_TRY
         const int size1 = TDenseSpaceType::Size1(rX);
         const int size2 = TDenseSpaceType::Size2(rX);
 
-        bool is_solved = true;
+        bool status = true;
 
         VectorType x(size1);
         VectorType b(size1);
 
-        // define an object to store skyline matrix and factorization
-        LUSkylineFactorization<TSparseSpaceType, TDenseSpaceType> myFactorization;
-        // copy myMatrix into skyline format
-        myFactorization.copyFromCSRMatrix(rA);
-        // factorize it
-        myFactorization.factorize();
-
-        for(int i = 0 ; i < size2 ; i++)
-        {
+        this->InitializeSolutionStep(rA, x, b);
+        for(int i = 0 ; i < size2 ; i++) {
             TDenseSpaceType::GetColumn(i,rX, x);
             TDenseSpaceType::GetColumn(i,rB, b);
-
-            // and back solve
-            myFactorization.backForwardSolve(size1, b, x);
-
+            status &= this->PerformSolutionStep(rA, x, b);
             TDenseSpaceType::SetColumn(i,rX, x);
             TDenseSpaceType::SetColumn(i,rB, b);
         }
+        this->FinalizeSolutionStep(rA, x, b);
 
-        return is_solved;
+        return status;
+        KRATOS_CATCH("")
     }
 
-
+    void FinalizeSolutionStep(SparseMatrixType& rLhs,
+                              VectorType& rSolution,
+                              VectorType& rRhs) override
+    {
+        mFactorization.clear();
```

Comment:
Is there a specific reason you do the clearing in the `FinalizeSolutionStep`, isn't it enough to do it in the `InitializeSolutionStep` before the factorization as you already do there. Then you may be able to avoid two times the clearing, which will reduce unnecessary operations.